src/dbnode/storage/series/buffer.go

```diff
@@ -537,7 +588,45 @@ func (b *dbBuffer) ReadEncoded(
 	return res, nil
 }
 
+func (b *dbBuffer) FetchBlocksForColdFlush(
+	ctx context.Context,
+	start time.Time,
+	version int,
+	nsCtx namespace.Context,
+) ([]xio.BlockReader, error) {
+	res := b.fetchBlocks(ctx, []time.Time{start},
+		streamsOptions{filterWriteType: true, writeType: ColdWrite, nsCtx: nsCtx})
+	if len(res) != 1 {
```

Comment:
What if it got evicted for falling out of retention or something? Also is it guaranteed to be 1 (not 2 or 3 if multiple encoder or w/e)

Reply:
Good point regarding falling out of retention - I don't think there is any other case where we won't get any data here.

`fetchBlocks` returns one result per block start, so this must be length 1 if we get any data.